crates/core/src/diff/fragment/tests.rs

```diff
@@ -1,7 +1,275 @@
+    let root: RootDiff = serde_json::from_str(initial_json).expect("Failed to deserialize fragment");
+    let root: Root = root.try_into().expect("Failed to convert RootDiff to Root");
+    let increment_diff = r#"{"0": "1"}"#;
+    let other_root : RootDiff = serde_json::from_str(increment_diff).expect("Failed to deserialize diff fragment");
+    let new_root = root.merge(other_root).expect("Failed to merge new root in");
+    let expected_root = r#"{
+        "0":"1",
+        "s":["\u003cScaffold\u003e\n  \u003cTopAppBar\u003e\n    \u003cTitle\u003e\u003cText\u003eHello\u003c/Text\u003e\u003c/Title\u003e\n  \u003c/TopAppBar\u003e\n  \u003cColumn width\u003d\"fill\" verticalArrangement\u003d\"center\" horizontalAlignment\u003d\"center\"\u003e\n    \u003cText style\u003d\"headlineLarge\"\u003eTitle\u003c/Text\u003e\n    \u003cCard shape\u003d\"8\" padding\u003d\"16\" width\u003d\"140\" height\u003d\"120\" elevation\u003d\"{\u0027defaultElevation\u0027: \u002710\u0027, \u0027pressedElevation\u0027: \u00272\u0027}\" phx-click\u003d\"dec\"\u003e\n      \u003cText padding\u003d\"16\"\u003eHello Jetpack!\u003c/Text\u003e\n    \u003c/Card\u003e\n    \u003cSpacer height\u003d\"8\"\u003e\u003c/Spacer\u003e\n    \u003cCard padding\u003d\"16\"\u003e\n      \u003cText padding\u003d\"16\"\u003eSimple card\u003c/Text\u003e\n    \u003c/Card\u003e\n    \u003cButton phx-click\u003d\"navigate\" contentPadding\u003d\"50\" elevation\u003d\"{\u0027defaultElevation\u0027: \u002720\u0027, \u0027pressedElevation\u0027: \u002710\u0027}\"\u003e\n      \u003cText\u003eNavigate to counter\u003c/Text\u003e\n    \u003c/Button\u003e\n    \u003cButton phx-click\u003d\"redirect\"\u003e\u003cText\u003eRedirect to counter\u003c/Text\u003e\u003c/Button\u003e\n    \u003cIconButton phx-click\u003d\"inc\" colors\u003d\"{\u0027containerColor\u0027: \u0027#FFFF0000\u0027, \u0027contentColor\u0027: \u0027#FFFFFFFF\u0027}\"\u003e\n      \u003cIcon imageVector\u003d\"filled:Add\"\u003e\u003c/Icon\u003e\n    \u003c/IconButton\u003e\n    \u003cRow verticalAlignment\u003d\"center\"\u003e\n      \u003cButton phx-click\u003d\"dec\" shape\u003d\"circle\" size\u003d\"60\"\u003e\n        \u003cText\u003e-\u003c/Text\u003e\n      \u003c/Button\u003e\n      \u003cText\u003eThis counter: ","\u003c/Text\u003e\n      \u003cButton phx-click\u003d\"inc\" shape\u003d\"circle\" size\u003d\"60\"\u003e\u003cText\u003e+\u003c/Text\u003e\u003c/Button\u003e\n    \u003c/Row\u003e\n  \u003c/Column\u003e\n\u003c/Scaffold\u003e"]}"#;
```

Comment:
Hmm, I've tried this and had no luck.

Reply:
Doing: 
```rust
    let initial_statics = r#"<Scaffold>
  <TopAppBar>
    <Title><Text>Hello</Text></Title>
  </TopAppBar>
  <Column width=\"fill\" verticalArrangement=\"center\" horizontalAlignment=\"center\">
    <Text style=\"headlineLarge\">Title</Text>
    <Card shape=\"8\" padding=\"16\" width=\"140\" height=\"120\" elevation=\"{'defaultElevation': '10', 'pressedElevation': '2'}\" phx-click=\"dec\">
      <Text padding=\"16\">Hello Jetpack!</Text>
    </Card>
    <Spacer height=\"8\"></Spacer>
    <Card padding=\"16\">
      <Text padding=\"16\">Simple card</Text>
    </Card>
    <Button phx-click=\"navigate\" contentPadding=\"50\" elevation=\"{'defaultElevation': '20', 'pressedElevation': '10'}\">
      <Text>Navigate to counter</Text>
    </Button>
    <Button phx-click=\"redirect\"><Text>Redirect to counter</Text></Button>
    <IconButton phx-click=\"inc\" colors=\"{'containerColor': '#FFFF0000', 'contentColor': '#FFFFFFFF'}\">
      <Icon imageVector=\"filled:Add\"></Icon>
    </IconButton>
    <Row verticalAlignment=\"center\">
      <Button phx-click=\"dec\" shape=\"circle\" size=\"60\">
        <Text>-</Text>
      </Button>
      <Text>This counter: ","</Text>
      <Button phx-click=\"inc\" shape=\"circle\" size=\"60\"><Text>+</Text></Button>
    </Row>
  </Column>
</Scaffold>"#;
    let initial_json = format!(r#"{{
        "0":"0",
        "s":["{}"]
    }}"#, initial_statics);
    let root: RootDiff = serde_json::from_str(&initial_json).expect("Failed to deserialize fragment");
```

results in `Failed to deserialize fragment: Error("control character (\\u0000-\\u001F) found while parsing a string", line: 4, column: 0)` which is annoying because it'd be so much cleaner. :/